cora/signal/lss.py

```diff
@@ -311,23 +305,24 @@ def process(self) -> InitialLSS:
         # NOTE: it is important not to set this any higher, otherwise power will alias
         # back down when the map is transformed later on
         lmax = 3 * self.nside - 1
+        nz = len(redshift)
+
+        # Create an extended covariance matrix capturing the cross correlations between
+        # the fields
+        cla = mpiarray.zeros((lmax + 1, 2 * nz, 2 * nz), axis=0)
 
         self.log.debug("Generating C_l(x, x')")
         cla0 = corrfunc.corr_to_clarray(corr0, lmax, xa, xromb=self.xromb, q=self.leg_q)
-        cla2 = corrfunc.corr_to_clarray(corr2, lmax, xa, xromb=self.xromb, q=self.leg_q)
-        cla4 = corrfunc.corr_to_clarray(corr4, lmax, xa, xromb=self.xromb, q=self.leg_q)
+        cla.local_array[:, nz:, nz:] = cla0
```

Comment:
Do we need `cla0.local_array` here, or is that redundant?

Reply:
It shouldn't matter. I actually don't think we even need `cla.local_array`, it should be happy to write one MPIArray into another. I'll do a quick test and change to be consistent